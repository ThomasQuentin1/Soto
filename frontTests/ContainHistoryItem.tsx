import React from "react";
import HistoryItem from "../components/history/HistoryItem";
import { Cart } from 'interfaces/Cart';

const ContainHistoryItem = () => {
  const cart : Cart[] = [{shop: {name: 'Auchan', city: 'Strasbourg', long: 1, lat: 1, id:1, server:'null', code:'null'}, price: 55, dateCreated: new Date(), dateLastEdit: new Date(), products: [{name: 'Pain', brand: 'Bon matin', scoreHealth: 20, quantity: 1, price: 1}, {name: 'Soda', brand: 'Coca-cola', scoreHealth: 1, quantity: 2, price: 1}, {name: 'Pâtes', brand: 'Barilla', scoreHealth: 100, quantity: 25, price: 2}, {name: 'Sauce tomate', brand: 'Panzani', scoreHealth: 50, quantity: 1, price: 1}]}, {shop: {name: 'Leclerc', city: 'Strasbourg', long: 1, lat: 1, id:1, server:'null', code:'null'}, price: 55, dateCreated: new Date(), dateLastEdit: new Date(), products: [{name: 'Pain', brand: 'Bon matin', scoreHealth: 20, quantity: 1, price: 1}, {name: 'Soda', brand: 'Coca-cola', scoreHealth: 1, quantity: 2, price: 1}, {name: 'Pâtes', brand: 'Barilla', scoreHealth: 100, quantity: 25, price: 2}, {name: 'Sauce tomate', brand: 'Panzani', scoreHealth: 50, quantity: 1, price: 1}]}];

  return <HistoryItem cart={cart[0]}></HistoryItem>;
};

export default ContainHistoryItem;
