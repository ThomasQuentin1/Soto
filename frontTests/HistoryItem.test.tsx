import React from 'react';
import renderer from 'react-test-renderer';
import ContainHistoryItem from './ContainHistoryItem';

test('ToggleLanguague is the components that display the button which change the language', () => {
    const component = renderer.create(
        <ContainHistoryItem></ContainHistoryItem>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});