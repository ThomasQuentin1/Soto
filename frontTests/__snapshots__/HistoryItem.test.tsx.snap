// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleLanguague is the components that display the button which change the language 1`] = `
<div
  className="MuiGrid-root history_item MuiGrid-item MuiGrid-grid-xs-6"
>
  <div
    className="MuiGrid-root MuiGrid-container"
    style={
      Object {
        "paddingLeft": "50px",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      style={
        Object {
          "marginBottom": "5px",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Mercredi 10 Mars
         - 
        Auchan
         
        Strasbourg
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Vous avez payé : 
          55
          €
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Total de produits commandés : 
          29
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
            onClick={[Function]}
          >
            Plus
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              style={
                Object {
                  "position": "absolute",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </p>
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            href="/shop"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <span
              className="MuiButton-label"
            >
              Ajouter au panier
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root"
  />
</div>
`;
